Source/Eto/Forms/Controls/Splitter.cs

```diff
@@ -87,11 +90,43 @@ protected virtual void OnPositionChanged(EventArgs e)
 
 		#endregion
 
+		/// <summary>
+		/// Initializes a new instance of the <see cref="T:Eto.Forms.Splitter"/> class.
+		/// </summary>
+		public Splitter()
+		{
+			shown = false;
+			panel1MinWidth = -1;
+			panel2MinWidth = -1;
+
+			Shown += Splitter_Shown;
+		}
+
+		private void Splitter_Shown(object o, EventArgs args)
+		{
+			if (!shown)
+			{
+				shown = true;
+				SizeChanged += (sender, e) => EnsurePosition();
+			}
+		}
+
 		static Splitter()
 		{
 			EventLookup.Register<Splitter>(c => c.OnPositionChanged(null), Splitter.PositionChangedEvent);
 		}
 
+		private void EnsurePosition()
+		{
+			if (Width <= 0)
```

Comment:
This method will have to be updated to deal with both horizontal and vertical orientations.